lib/resourcemerge/machineconfig.go

```diff
@@ -78,6 +79,37 @@ func ensureControllerConfigSpec(modified *bool, existing *mcfgv1.ControllerConfi
 	setBytesIfSet(modified, &existing.KubeAPIServerServingCAData, required.KubeAPIServerServingCAData)
 	setBytesIfSet(modified, &existing.CloudProviderCAData, required.CloudProviderCAData)
 
+	// First off: do I need to generally account for Infra changes? Seems like we should?
+	if required.Infra != nil && !equality.Semantic.DeepEqual(existing.Infra, required.Infra) {
+		*modified = true
+		existing.Infra = required.Infra
+	}
+
+	// Moving on... if all Infras are empty (ie born in 4.1 clusters), maybe update Infra with values we _do_ have(?)
+	if required.Infra == nil && existing.Infra == nil {
+		// from types.go: "platform is deprecated, use Infra.Status.PlatformStatus.Type instead"
+		if existing.Platform != "" {
+			*modified = true
+			existing.Infra = &configv1.Infrastructure{
+				Status: configv1.InfrastructureStatus{
+					PlatformStatus: &configv1.PlatformStatus{
+						// but errr do I need to make sure this platform is the correct constant (as seen in types_infrastructure.go)???
+						Type: configv1.PlatformType(existing.Platform),
+					},
+				},
+			}
+		}
+		// but alsooo from types.go: "etcdDiscoveryDomain is deprecated, use Infra.Status.EtcdDiscoveryDomain instead"
+		if existing.EtcdDiscoveryDomain != "" {
+			*modified = true
+			required.Infra = &configv1.Infrastructure{
+				Status: configv1.InfrastructureStatus{
+					EtcdDiscoveryDomain: existing.EtcdDiscoveryDomain,
+				},
+			}
+		}
+	}
```

Comment:
I'm thinking the infra sync at line 83 should just be enough :thinking: when this code runs the MCO sync function makes sure we have a "good" `required` infra object in the CC and that should be enough to avoid the panic and keep this working and in sync - a live test on a cluster coming from 4.1 will confirm this :woman_dancing:

Reply:
I.E. if we don't have anything at the API level for CC.infra (cause it's nil and never sync), it should be just enough to sync it here with required (which we populate in sync.go) and that's it

Reply:
done